README.md

```diff
@@ -254,8 +254,135 @@ class MysqlSpec extends FlatSpec with ForAllTestContainer {
 }
 ```
 
+## New API
+
+Starting from 0.34.0 version testcontainers-scala provides the new API. 
+The main motivation points are in the [pull request](https://github.com/testcontainers/testcontainers-scala/pull/78).
+
+### `Container` and `ContainerDef` 
+
+Docker containers are represented through the two different entities:
+1. `ContainerDef` — it's container definition. `ContainerDef` describes, how to build a container.
+   You can think about it like about a container constructor, or dockerfile description.
+   Usually, `ContainerDef` receives some parameters.
+   `ContainerDef` has a `start()` method. It returns a started `Container`.
+2. `Container` — it's a started container. You can interact with it through its methods.
+   For example, in the case of `MySQLContainer` you can get it's JDBC URL with `jdbcUrl` method.
+   `Container` is the main entity for using inside tests.
+
+### Scalatest usage
+
+You can use one of the four traits:
+1. `TestContainerForAll` — will start a single container before all tests and stop after all tests.
+2. `TestContainerForEach` — will start a single container before each test and stop after each test.
+3. `TestContainersForAll` — will start multiple containers before all tests and stop after all tests.
+4. `TestContainersForEach` — will start multiple containers before each test and stop after each test.
+
+
+#### Single container in tests
+
+If you want to use a single container in your test:
+```scala
+class MysqlSpec extends FlatSpec with TestContainerForAll {
+
+  // You need to override `containerDef` with needed container definition
+  override val containerDef = MySQLContainer.Def()
+
+  // To use containers in tests you need to use `withContainers` function
+  it should "test" in withContainers { mysqlContainer =>
+    // Inside your test body you can do with your container whatever you want to
+    assert(mysqlContainer.jdbcUrl.nonEmpty)
+  }
+}
+```
+
+Usage of `TestContainerForEach` is not different from the example above.
+
+#### Multiple containers in tests
+
+If you want to use multiple containers in your test:
+```scala
+class ExampleSpec extends FlatSpec with TestContainersForAll {
+
+  // First of all, you need to declare, which containers you want to use
+  override type Containers = MySQLContainer and PostgreSQLContainer
+
+  // After that, you need to describe, how you want to start them,
+  // In this method you can use any intermediate logic.
+  // You can pass parameters between containers, for example.
+  override def startContainers(): Containers = {
```

Comment:
1) is it possible to start the containers under the hood? The order of start can be inherited from the order of returned definitions
2) is it possible to add default behavior for cases with no additional configuration (like this one in the doc)?

Reply:
I tried to find a way to do this and I came to the conclusion that there is no need for this _default_ behavior. First of all, you need to pass parameters inside the container's definitions. So, anyway, you need to call `ContainerDef` constructor. Where will you do this? Even if you do this inside a test body (like in the classic API) you will need the same amount of code.

Let's think about it. Imagine, that we have some _default_ way to start containers in a defined order. I can imagine API like that:
```scala
class ExampleSpec extends FlatSpec with TestContainersForAll {

  val containers = MySQLContainer.Def(paramsHere) and PostgreSQLContainer.Def(paramsHere)
  
  // Tests here
}
```
And it looks exactly like the current API! With all its advantages and disadvantages. The main problem with it — how will you express dependent containers? Looks like you need a code block with the initialization logic:
```scala
class ExampleSpec extends FlatSpec with TestContainersForAll {

  val containers = {
    val c1 = Container1.Def()
    val c2 = Container2.Def(c1.someParam)
    c1 and c2
  }
  
  // Tests here
}
```
But, this code will not work, because `c1.someParam` will not be accessible, because you can get some parameters only from started containers. So, you need to start them:
```scala
class ExampleSpec extends FlatSpec with TestContainersForAll {

  val containers = {
    val c1 = Container1.Def().start()
    val c2 = Container2.Def(c1.someParam).start()
    c1 and c2
  }
  
  // Tests here
}
```
But you can't create a `val` inside your test body, because `val` is eager and will start to initialize with the test constructor. So, if you want to control containers initialization time, you need to make it `def`:
```scala
class ExampleSpec extends FlatSpec with TestContainersForAll {

  def startContainers = {
    // Initialization here
  }
  
  // Tests here
}
```
But, how will you provide started containers to the user? It looks like you need some function that will know about startup logic. Like `withContainers`:
```scala
class ExampleSpec extends FlatSpec with TestContainersForAll {

  def startContainers = {
    // Initialization here
  }
  
  it should "test" in withContainers { c1 and c2 =>
    // Test body
  }
}
```
`withContainers` should receive a function from `containers` type to a Unit. But what is `containers` type exactly? If you want to provide reusable traits like `TestContainersForAll` you need to abstract from this type somehow. And this is exactly what `Containers` type do. And initialization block — is the `startContainers` method. And eventually we end up with the same API:
```scala
class ExampleSpec extends FlatSpec with TestContainersForAll {

  override type Containers = Container1 and Container2

  override def startContainers(): Containers = {
    val c1 = Container1.Def().start()
    val c2 = Container2.Def(c1.someParam).start
    c1 and c2
  }
  
  it should "test" in withContainers { c1 and c2 =>
    // Test body
  }
}
```

Also, if you don't have dependent containers, you can write a bit less verbosely:
```scala
def startContainers() = Container1.Def().start() and Container2.Def().start()
```
Looks decent for me.

Actually, the reasoning above is the exact reasoning of this whole API. This is how I end up with this API.